src/content/platform-integration/web/wasm.md

```diff
@@ -79,123 +37,93 @@ that has been migrated to be
 
-### Build or run the app `flutter build web --wasm`
+### Run or build your app
+
+To run the app with Wasm for development or testing,
+use the `--wasm` flag with the `flutter run command.
+
+```console
+$ flutter run -d chrome --wasm
+```
 
 To build a web application with Wasm, add the `--wasm` flag to
 the existing `flutter build web` command.
 
 ```console
-flutter build web --wasm
+$ flutter build web --wasm
 ```
 
 The command produces output into the `build/web` directory relative to the
 package root, just like `flutter build web`.
 
-Wasm is also supported by the `flutter run` command on Flutter 3.24 or newer.
-If running the app this way, Flutter serves the app locally as it would
-normally, and the next step is not required.
-
-:::note
+### Open the app in a compatible web browser
 Even with the `--wasm` flag, Flutter will still compile the application to
 JavaScript. If WasmGC support is not detected at runtime, the JavaScript output
 is used so the application will continue to work across
-browsers.
-:::
 
-### Serve the built output with an HTTP server
-
-Flutter web WebAssembly uses multiple threads to render your application
-faster, with less jank. To do this, advanced browser features are used that
-require specific HTTP response headers.
-
-:::warning
-Flutter web applications won't run with WebAssembly unless the server is
-configured to send specific HTTP headers.
-:::
-
-| Name | Value |
-|-|-|
-| `Cross-Origin-Embedder-Policy` | `credentialless` <br> or <br> `require-corp` |
-| `Cross-Origin-Opener-Policy` | `same-origin` |
-
-{:.table}
-
-To learn more about these headers, check out
-[Load cross-origin resources without CORP headers using COEP: credentialless][coep].
-
-:::tip Serving locally with `dhttpd`
-To view the built output locally, you can use the
-[`dhttpd` package]({{site.pub-pkg}}/dhttpd).
-
-After installation, run it from the 'build/web' directory, and use the following
-command to add the necessary headers:  
-`dhttpd '--headers=Cross-Origin-Embedder-Policy=credentialless;Cross-Origin-Opener-Policy=same-origin'`
-:::
-
-[coep]: https://developer.chrome.com/blog/coep-credentialless-origin-trial
-
-### Load it in a browser
-
-As of {{last-update}},
-[only **Chromium-based browsers**](#chrome-119-or-later)
-(Version 119 or later) are able to run Flutter/Wasm content.
-
-You can verify whether the app is actually running with Wasm in two ways:
+You can verify whether the app is actually running with Wasm by checking for 
+the `dart2wasm` environment variable, set during compilation (preferred).
 
 ```dart
-// Using the environment variable set by compilation (preferred)
 const isRunningWithWasm = bool.fromEnvironment('dart.tool.dart2wasm');
-
-// Using differences in number representation to test whether native (Wasm)
-// number representation is used
-final isRunningWithWasm = identical(double.nan, double.nan);
 ```
 
-## Known limitations
+Alternatively, you can use differences in number representations
```

Comment:
As re-written, how is this a limitation? Can you make it more clear?

Reply:
This section isn't about limitations, it's about detecting if your app is running with WebAssembly. I moved the "known limitations" section further down the page.